### ISSUE TITLE
python.envFile has no effect

### ISSUE THREAD

**Issue by mijamo** [DonJayamanne/pythonVSCode#1245]
## Environment data
VS Code version:  1.16.1
Python Extension version:  0.7.0
Python Version:   2.7.12
OS and version: Ubuntu 16.04

## Actual behavior

`python.envFile` is not used in any way.

## Expected  behavior

The file in `python.envFile` should be read and each line should be loaded as an env variable for Python.

## Steps to reproduce:  
-  Create an `.env`file, for instance in `.vscode/.env`
-  Write some variables into it, ex: `MYVAR="dude"`
- Write a simple test Python file:
```python
import os

print os.environ["MYVAR"]
```
- Run the file with python (right click -> run Python File in terminal)
- Get the error `KeyError: 'MYVAR'` showing that the variable is indeed not seen by Python

This is annoying as there is no way to set the PYTHONPATH variable as of today, so I cannot just run nosetest through the regular interface as I need to be able to set PYTHONPATH to include other libraries.

**Comment by DonJayamanne** [comment]
Environment variables defined in .env are read only when debugging and running unit tests. 
Running something in a VS Code terminal is just the same as running the python file in an external terminal (& to my knowledge these files aren't automatically read by python).

Also, please note that changes to .env file will not be reflected untill and unless VS Code is closed & re-opened (#1230)

**Comment by mijamo** [comment]
Mh alright. However it does seem that the workspace `"python.envFile"` is ignored by the launch configuration, so I need to specify the env file in each launch config separately. Is that also expected?

I think the "envFile" property for the launch config should default to the one in the workspace if nothing is specified.

Currently, when I have this config in launch it fails:
```
{
         "name": "Python",
         "type": "python",
         "request": "launch",
         "stopOnEntry": true,
         "pythonPath": "${config:python.pythonPath}",
         "program": "${file}",
         "cwd": "${workspaceRoot}",
         "env": {},
         "debugOptions": [
            "WaitOnAbnormalExit",
            "WaitOnNormalExit",
            "RedirectOutput"
         ]
      }
```

But it works with this one:
```
{
         "name": "Python",
         "type": "python",
         "request": "launch",
         "stopOnEntry": true,
         "pythonPath": "${config:python.pythonPath}",
         "program": "${file}",
         "cwd": "${workspaceRoot}",
         "env": {},
         "envFile": "${workspaceRoot}/.vscode/.env",
         "debugOptions": [
            "WaitOnAbnormalExit",
            "WaitOnNormalExit",
            "RedirectOutput"
         ]
      }
```

My settings are:
```
{
   "python.pythonPath": "myPythonExecPath",
   "python.envFile": "${workspaceRoot}/.vscode/.env",
}
```

So basically `"python.envFile"` seems to be used ONLY but unittests right now, not when debugging, because for that we need to manually respecify it.

I would this this is a bug, but maybe I'm wrong here.

**Comment by DonJayamanne** [comment]
@mijamo Agreed, it needs to be the default setting. Will make the necessary changes for the next release. Thanks for pointing this out.

**Comment by DonJayamanne** [comment]
@mijamo It is the default setting when you start debugging. The problem may have been the fact that you have an old launch.json setting.

**Comment by mijamo** [comment]
Sorry to still bother but I still think there is something wrong here : 
- If you first set the workspace setting "python.envFile" and then create a luanch config, the new launch configuration contains `"envFile": "${workspaceRoot}/.env"`, no matter what I specified in the workspace settings.
- If you remove the `"envFile"` setting from a launch configuration, no env file is loaded when you debug.

I tried with a brand new project with a brand new launch.json with Python 0.7.0 installed and confirm the behavior: `"python.envFile"` is NEVER used by the debugger in any configuration I have found.

I would except the following behavior:
- No `"envFile"` setting by default in any launch configuration
- All launch configuration use `"python.envFile"` if they don't have a `"envFile"` setting specified.

**Comment by DonJayamanne** [comment]
@mijamo 
I'm a little confused by what you've written, here's how it works:
1. Debugging will not using what is in your workspace settings when dealing with env variables or env files.  
2. If you have a setting ```envFile``` in launch.json, it will be used.  
3. If you do not have a setting ```envFile```` in launch.json it will not be used.

Don't confuse your self with what's in the workspace settings.json (that's used for intellisense, linters and the like).

**Comment by Euruson** [comment]
Well, I found this may explain the problem.
> [The predefined variables are not supported in strings in settings.json files. ](https://code.visualstudio.com/docs/editor/variables-reference)

So `"python.envFile": "${workspaceFolder}/.vscode/xxx.env"` in settings.json and `"envFile": "${config:python.envFile}"` in launch.json will not work. But `"envFile": "${workspaceFolder}/.vscode/xxx.env"` in launch.json can load the .env file.

I think `"python.envFile": "${workspaceFolder}/.env"` in default settings is really confusing. This setting may only be used in unittests, but not debbugging, no matter how to modify launch.json with `python.envFile`.
